pkg/reconciler/pipelinerun/resources/pipelinerunresolution.go

```diff
@@ -376,6 +377,34 @@ func GetPipelineConditionStatus(pr *v1alpha1.PipelineRun, state PipelineRunState
 	}
 }
 
+// GetPipelineCompletionTime will return the CompletionTime that the PipelineRun pr should be
+// updated with, based on the status and CompletionTime of the TaskRuns in state.
+func GetPipelineCompletionTime(pr *v1alpha1.PipelineRun, state PipelineRunState, logger *zap.SugaredLogger, dag *dag.Graph) *metav1.Time {
```

Comment:
This seems like it should have a test.

Reply:
Definitely, once we settle on the overall design I'll be happy to add tests. Right now I'm still looking for feedback from someone familiar with the code and any implications surrounding CompletionTime on whether this change makes sense.